File: server/MotoRevApi/Model/Alerta.cs
using MotoRevApi.Enums;

namespace MotoRevApi.Model;

public class Alerta
{
    public int Id { get; set; }
    public TipoAlerta Tipo { get; set; }
    public string UsuarioId { get; set; } = string.Empty;
    public bool Lido { get; set; } = false;
    public DateTime CriadoEm { get; set; } = DateTime.UtcNow;
    public int? MotoId { get; set; }
    public int? AgendamentoId { get; set; }
    public int? Quilometragem { get; set; }
}

File: client/src/app/models/Alerta.ts
export type TipoAlerta =
  | 'RevisaoProxima'
  | 'RevisaoAtrasada'
  | 'AgendamentoCriado'
  | 'AgendamentoAlterado'
  | 'RevisaoConcluida'
  | 'AgendamentoAprovado'
  | 'AgendamentoRecusado'
  | 'NovaSolicitacao'
  | 'Cancelamento'
  | 'Reagendamento';

export interface AlertaResponse {
  id: number;
  tipo: TipoAlerta;
  lido: boolean;
  criadoEm: string;
  motoId?: number;
  agendamentoId?: number;
  quilometragem?: number;
  ordemRevisao?: number;
  modeloMotoNome?: string;
  marcaMoto?: string;
}

export interface AlertaNaoLidosCountResponse {
  total: number;
}
